tests/unit/test_resolver.py

```diff
@@ -172,43 +172,113 @@ def _resolve_packages(request: Request) -> RequestOutput:
 
 
 @pytest.mark.parametrize(
-    "flags",
-    [
-        pytest.param(["dev-package-managers"], id="dev-package-managers-true"),
-        pytest.param([], id="dev-package-managers-false"),
-    ],
+    "package_type",
+    ["unknown", "nonexistent"],
 )
-def test_dev_mode(flags: list[str], tmp_path: Path) -> None:
-    mock_resolver = mock.Mock()
-    mock_resolver.return_value = RequestOutput.empty()
+def test_completely_unknown_package_manager_error(
+    package_type: str,
+    tmp_path: Path,
+) -> None:
+    """Test that completely unknown package managers raise proper error."""
     with (
         mock.patch.dict(
             resolver._package_managers,
-            values={"gomod": mock_resolver},
+            values={"gomod": mock.Mock(return_value=RequestOutput.empty())},
             clear=True,
         ),
         mock.patch.dict(
             resolver._dev_package_managers,
-            values={"shrubbery": mock_resolver},
+            values={},
             clear=True,
         ),
+        mock.patch(
+            "hermeto.core.models.input.get_args",
+            return_value=("bundler", "cargo", "generic", "gomod", "npm", "pip", "rpm", "yarn"),
+        ),
     ):
-        dev_package_input = mock.Mock()
-        dev_package_input.type = "shrubbery"
+        package_input = mock.Mock()
+        package_input.type = package_type
 
         request = mock.Mock()
         request.source_dir = RootedPath(tmp_path)
-        request.flags = flags
-        request.packages = [dev_package_input]
+        request.flags = []
+        request.packages = [package_input]
+
+        expected_error = re.escape(f"Package manager(s) not yet supported: {package_type}")
+        with pytest.raises(UnsupportedFeature, match=expected_error):
+            resolver.resolve_packages(request)
 
-        if flags:
-            assert resolver.resolve_packages(request) == RequestOutput(
-                components=[], build_config=BuildConfig(environment_variables=[], project_files=[])
-            )
-        else:
-            expected_error = re.escape("Package manager(s) not yet supported: shrubbery")
-            with pytest.raises(UnsupportedFeature, match=expected_error):
-                resolver.resolve_packages(request)
+
+def test_experimental_package_used_without_x_prefix_suggests_x_prefix(tmp_path: Path) -> None:
+    """Test that using experimental package without x-prefix suggests the x-prefix version."""
+    from typing import get_args
+
+    from hermeto.core.errors import InvalidInput
+    from hermeto.core.models.input import PackageManagerType, Request, parse_user_input
+
+    # Mock get_args to return the experimental type when called
+    original_args = get_args(PackageManagerType)
+    with mock.patch(
+        "hermeto.core.models.input.get_args", return_value=original_args + ("x-shrubbery",)
+    ):
+        # Create input data that will trigger input validation
```

Comment:
A minor nitpick: could you please move this initialization out of `with mock` clause? It does not look like it s necessary to be present in that scope.